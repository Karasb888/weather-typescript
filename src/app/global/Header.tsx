import * as React from 'react';
import CityForm from './CityForm';

const Header: React.FC = () => {
    return (
        <header>
            <CityForm />
        </header>
    );
};

export default Header;
