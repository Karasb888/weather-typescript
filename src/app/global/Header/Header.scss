.header {
    display: flex;
    justify-content: center;
    padding: 25px 0;
}

.homeBtn {
    cursor: pointer;
    font-size: 25px;
    position: absolute;
    left: 10px;
    top: 10px;
    text-transform: uppercase;
    font-family: 'Cooper-Hewitt';
    font-weight: 600;
    letter-spacing: 3px;
    color: #157571;
    &:hover {
        text-decoration: underline;
    }
}