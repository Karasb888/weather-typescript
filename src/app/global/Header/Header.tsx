import * as React from 'react';
import SearchForm from '../SearchForm/SearchForm';
import * as styles from './Header.scss';
import { History } from 'history';
import { withRouter } from 'react-router-dom';

interface Props {
    history: History
};

const Header: React.FC<Props> = ({ history }) => {
    return (
        <div className={styles.header}>
            <div className={styles.homeBtn} onClick={() => { history.push('/') }}>home</div>
            <SearchForm />
        </div>
    );
};

export default withRouter(Header);
