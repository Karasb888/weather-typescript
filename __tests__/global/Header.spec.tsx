import * as React from 'react';
import { shallow } from 'enzyme';
import Header from '../../src/app/global/Footer/Footer';

describe('++++ <Header /> ++++', () => {
    it('Renders correctly', () => {
        const wrapper = shallow(<Header />);
        expect(wrapper).toMatchSnapshot();
    });
});
