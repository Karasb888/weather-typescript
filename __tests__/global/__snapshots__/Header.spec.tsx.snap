// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`++++ <Header /> ++++ Renders correctly 1`] = `
<div
  className="footer"
>
  Weather data provided by 
  <a
    href="https://openweathermap.org/"
    rel="noopener noreferrer"
    target="_blank"
  >
    https://openweathermap.org/
  </a>
</div>
`;
